
project contrail-web-core/
diff --git a/dev-install.sh b/dev-install.sh
index 18d82edd..cb838031 100755
--- a/dev-install.sh
+++ b/dev-install.sh
@@ -308,6 +308,6 @@ cp -af ./$THIRD_PARTY/jquery-linedtextarea.css webroot/assets/jquery-linedtextar
 rm -f webroot/css/contrail.unified.css
 
 # compile sass
-./node_modules/node-sass/bin/node-sass webroot/common/ui/scss/contrail.thirdparty.unified.scss > webroot/common/ui/css/contrail.thirdparty.unified.css
-./node_modules/node-sass/bin/node-sass webroot/common/ui/scss/contrail.unified.scss > webroot/common/ui/css/contrail.unified.css
+#./node_modules/node-sass/bin/node-sass webroot/common/ui/scss/contrail.thirdparty.unified.scss > webroot/common/ui/css/contrail.thirdparty.unified.css
+#./node_modules/node-sass/bin/node-sass webroot/common/ui/scss/contrail.unified.scss > webroot/common/ui/css/contrail.unified.css
 #End - Merging ALL CSS files
diff --git a/package.json b/package.json
index a16d3ae7..0baa40df 100644
--- a/package.json
+++ b/package.json
@@ -29,6 +29,7 @@
     "xml2js": "0.2.3",
     "redis-rstream": "0.1.2",
     "connect-redis": "1.4.5",
+    "async-foreach": "0.1.3",
     "request-timeout": "0.0.2"
   },
   "devDependencies": {

project controller/
diff --git a/src/analytics/OpServerProxy.cc b/src/analytics/OpServerProxy.cc
index ae4a8ae89..eb3614246 100644
--- a/src/analytics/OpServerProxy.cc
+++ b/src/analytics/OpServerProxy.cc
@@ -377,7 +377,7 @@ class OpServerProxy::OpServerImpl {
             // If redis returns error for async request, then perhaps it
             // is busy executing a script and it has reached the maximum
             // execution time limit.
-            BOOST_ASSERT_MSG(reply->type != REDIS_REPLY_ERROR, reply->str);
+            //BOOST_ASSERT_MSG(reply->type != REDIS_REPLY_ERROR, reply->str);
 
             if (rpi) {
                 rpi->ProcessCallback(reply);
diff --git a/src/analytics/syslog_collector.cc b/src/analytics/syslog_collector.cc
index bd6c84409..0d33d77de 100644
--- a/src/analytics/syslog_collector.cc
+++ b/src/analytics/syslog_collector.cc
@@ -16,7 +16,7 @@
 #include <boost/spirit/include/phoenix_core.hpp>
 #include <boost/spirit/include/phoenix_operator.hpp>
 #include <boost/spirit/include/phoenix_stl.hpp>
-#include <boost/spirit/home/phoenix/object/construct.hpp>
+#include <boost/phoenix/object/construct.hpp>
 #include <boost/uuid/uuid.hpp>
 #if __GNUC_PREREQ(4, 6)
 #pragma GCC diagnostic push
diff --git a/src/ifmap/ifmap_node.cc b/src/ifmap/ifmap_node.cc
index 1e56ec44b..70eba1b89 100644
--- a/src/ifmap/ifmap_node.cc
+++ b/src/ifmap/ifmap_node.cc
@@ -7,6 +7,7 @@
 #include <pugixml/pugixml.hpp>
 
 #include "ifmap/ifmap_table.h"
+#include <iostream>
 
 using namespace std;
 using pugi::xml_node;
diff --git a/src/zookeeper/zookeeper_client.cc b/src/zookeeper/zookeeper_client.cc
index 2a83698b0..a193a039b 100644
--- a/src/zookeeper/zookeeper_client.cc
+++ b/src/zookeeper/zookeeper_client.cc
@@ -9,6 +9,7 @@
 #include <zookeeper/zookeeper_client.h>
 #include <zookeeper/zookeeper_client_impl.h>
 #include <zookeeper/zookeeper_interface.h>
+#include <cstring>
 
 #define ZOO_LOG(_Level, _Msg)                                             \
     do {                                                                  \

project third_party/
diff --git a/packages.xml b/packages.xml
index 7481cbd..f227de0 100644
--- a/packages.xml
+++ b/packages.xml
@@ -42,7 +42,7 @@
 	  <version>20+</version>
 	</distribution>
 	<distribution>
-	  <name>suse</name>
+	  <name>SuSe</name>
 	  <version>12+</version>
 	</distribution>
       </exclude>
@@ -71,7 +71,7 @@
 	  <version>20+</version>
 	</distribution>
 	<distribution>
-	  <name>suse</name>
+	  <name>SuSe</name>
 	  <version>12+</version>
 	</distribution>
       </exclude>
@@ -97,7 +97,7 @@
 	  <version>20+</version>
 	</distribution>
 	<distribution>
-	  <name>suse</name>
+	  <name>SuSe</name>
 	  <version>12+</version>
 	</distribution>
       </exclude>
@@ -120,7 +120,7 @@
 	  <version>20+</version>
 	</distribution>
 	<distribution>
-	  <name>suse</name>
+	  <name>SuSe</name>
 	  <version>12+</version>
 	</distribution>
       </exclude>
@@ -142,7 +142,7 @@
 	  <version>20+</version>
 	</distribution>
 	<distribution>
-	  <name>suse</name>
+	  <name>SuSe</name>
 	  <version>12+</version>
 	</distribution>
       </exclude>
@@ -177,7 +177,7 @@
 	  <version>20+</version>
 	</distribution>
 	<distribution>
-	  <name>suse</name>
+	  <name>SuSe</name>
 	  <version>12+</version>
 	</distribution>
       </exclude>
@@ -274,7 +274,7 @@
 	  <version>14.04+</version>
 	</distribution>
 	<distribution>
-	  <name>suse</name>
+	  <name>SuSe</name>
 	  <version>12+</version>
 	</distribution>
       </exclude>
@@ -306,7 +306,7 @@
 	  <version>20+</version>
 	</distribution>
 	<distribution>
-	  <name>suse</name>
+	  <name>SuSe</name>
 	  <version>12+</version>
 	</distribution>
       </exclude>
@@ -351,7 +351,7 @@
 	  <version>20+</version>
 	</distribution>
 	<distribution>
-	  <name>suse</name>
+	  <name>SuSe</name>
 	  <version>12+</version>
 	</distribution>
       </exclude>

project tools/packages/
diff --git a/rpm/contrail/contrail.spec b/rpm/contrail/contrail.spec
index 19bca28..6163ee0 100644
--- a/rpm/contrail/contrail.spec
+++ b/rpm/contrail/contrail.spec
@@ -1,7 +1,7 @@
 %define         _unpackaged_files_terminate_build 0
 %define         _contrailetc /etc/contrail
 %define         _contrailutils /opt/contrail/utils
-%define         _distropkgdir tools/packaging/common/control_files
+%define         _distropkgdir /tools/packages/rpm/contrail
 %define         _contraildns /etc/contrail/dns
 %define         _distrorpmpkgdir tools/packages/rpm/contrail
 
@@ -23,7 +23,7 @@
 %if 0%{?_kVers:1}
 %define         _kvers      %{_kVers}
 %else
-%define         _kvers      3.10.0-229.el7.x86_64 3.10.0-327.10.1.el7.x86_64
+%define         _kvers      4.4.59-92.17-default
 %endif
 
 %{echo: "Building release %{_relstr}\n"}
@@ -62,7 +62,7 @@ BuildRequires:  python-lxml
 BuildRequires:  python-setuptools
 BuildRequires:  unzip
 BuildRequires:  vim-common
-BuildRequires:  protobuf
+#BuildRequires:  protobuf
 BuildRequires:  protobuf-compiler
 BuildRequires:  protobuf-devel
 BuildRequires:  net-snmp-python
@@ -115,10 +115,10 @@ python %{_sbtop}/tools/packages/utils/generate_doc_index.py %{buildroot}/usr/sha
 
 # Install supervisor files
 pushd %{_builddir}/..
-install -p -m 755 %{_distrorpmpkgdir}/supervisor-control.initd  %{buildroot}/etc/init.d/supervisor-control
-install -p -m 755 %{_distrorpmpkgdir}/supervisor-config.initd  %{buildroot}/etc/init.d/supervisor-config
-install -p -m 755 %{_distrorpmpkgdir}/supervisor-analytics.initd  %{buildroot}/etc/init.d/supervisor-analytics
-install -p -m 755 %{_distrorpmpkgdir}/supervisor-vrouter.initd  %{buildroot}/etc/init.d/supervisor-vrouter
+install -p -m 755 %{_sbtop}/tools/packages/rpm/contrail/supervisor-control.initd  %{buildroot}/etc/init.d/supervisor-control
+install -p -m 755 %{_sbtop}/tools/packages/rpm/contrail/supervisor-config.initd  %{buildroot}/etc/init.d/supervisor-config
+install -p -m 755 %{_sbtop}/tools/packages/rpm/contrail/supervisor-analytics.initd  %{buildroot}/etc/init.d/supervisor-analytics
+install -p -m 755 %{_sbtop}/tools/packages/rpm/contrail/supervisor-vrouter.initd  %{buildroot}/etc/init.d/supervisor-vrouter
 popd
 
 # Install contrail utilities
@@ -276,7 +276,7 @@ The VRouter Agent API is used to inform the VRouter agent of the association bet
 %{python_sitelib}/discoveryclient*
 %{python_sitelib}/libpartition*
 %{python_sitelib}/pysandesh*
-%{python_sitelib}/sandesh-0.1dev*
+%{python_sitelib}/sandesh-0.1.dev0*
 %{python_sitelib}/sandesh_common*
 %{python_sitelib}/vnc_api*
 %config(noreplace) %{_contrailetc}/vnc_api_lib.ini
@@ -527,7 +527,7 @@ chown -R contrail:adm /var/log/contrail
 chmod 0750 /var/log/contrail
 chown -R contrail:contrail /var/lib/contrail/ /etc/contrail/
 chmod 0750 /etc/contrail/
-chmod +x /etc/init.d/ifmap
+#chmod +x /etc/init.d/ifmap
 chmod +x /etc/init.d/contrail-api
 chmod +x /etc/init.d/contrail-schema
 chmod +x /etc/init.d/contrail-device-manager
@@ -541,7 +541,7 @@ Group:              Applications/System
 Requires:           xmltodict >= 0.7.0
 Requires:           contrail-lib >= %{_verstr}-%{_relstr}
 Requires:           python-pycassa
-Requires:           redis-py > 0.1-2contrail
+Requires:           redis-py 
 Requires:           redis >= 2.6.13-1
 Requires:           python-contrail >= %{_verstr}-%{_relstr}
 Requires:           python-psutil >= 0.6.0

project tools/packaging/
diff --git a/common/rpm/Makefile b/common/rpm/Makefile
index b5cfe459..bbdc6e9c 100644
--- a/common/rpm/Makefile
+++ b/common/rpm/Makefile
@@ -15,7 +15,8 @@ SB_TOP              := $(CPLANE_GIT_ROOT)/..
 SB_TOPDIR	    := --define '_sbtop $(SB_TOP)'
 PREP = prep
 PPREP = pprep
-
+TAG = SUSE
+CONTRAIL_SKU = 'juno'
 # Skip unit-tests for building packages
 export BUILD_ONLY := TRUE
 
diff --git a/common/rpm/contrail-api-lib.spec b/common/rpm/contrail-api-lib.spec
index 4946da53..12568d87 100644
--- a/common/rpm/contrail-api-lib.spec
+++ b/common/rpm/contrail-api-lib.spec
@@ -108,7 +108,7 @@ popd
 %{python_sitelib}/pysandesh
 %{python_sitelib}/sandesh_common
 %{python_sitelib}/sandesh_common-*
-%{python_sitelib}/sandesh-0.1dev*
+%{python_sitelib}/sandesh-0.1.dev0*
 %{_api_venv_archv}/vnc_api-0.1dev.tar.gz
 %{_anl_venv_archv}/vnc_api-0.1dev.tar.gz
 %{_api_venv_archv}/cfgm_common-0.1dev.tar.gz
diff --git a/common/rpm/contrail-database-common.spec b/common/rpm/contrail-database-common.spec
index fa126b50..7f2a3ee4 100644
--- a/common/rpm/contrail-database-common.spec
+++ b/common/rpm/contrail-database-common.spec
@@ -32,7 +32,7 @@ BuildArch: noarch
 
 Requires: contrail-database >= %{_verstr}-%{_relstr}
 Requires: sysstat
-Requires: datastax-agent
+#Requires: datastax-agent
 
 # Packaging zookeeper-3.4.8-0contrail1 for all rhel version
 %if "%{_rhel}" == "rhel"
diff --git a/common/rpm/contrail-database.spec b/common/rpm/contrail-database.spec
index df16ec72..e01e3571 100644
--- a/common/rpm/contrail-database.spec
+++ b/common/rpm/contrail-database.spec
@@ -32,7 +32,7 @@ BuildArch: noarch
 %if 0%{?el6}
 Requires: cassandra12
 %else
-Requires: cassandra22 >= 2.2.8
+Requires: cassandra
 %endif
 Requires: supervisor
 Requires: java-1.7.0-openjdk
@@ -89,25 +89,25 @@ fi
 #install database
 install -d -m 755 %{buildroot}%{_venv_root}
 pushd %{_builddir}/..
-tar zxvf %{_build_dist}/analytics/database/dist/database-0.1dev.tar.gz
-pushd database-0.1dev
+tar zxvf %{_build_dist}/analytics/database/dist/database-0.1.dev0.tar.gz
+pushd database-0.1.dev0
 %{__python} setup.py install --root=%{buildroot} %{?_venvtr}
 popd
-tar zxf %{_build_dist}/sandesh/common/dist/sandesh-common-0.1dev.tar.gz
-pushd sandesh-common-0.1dev
+tar zxf %{_build_dist}/sandesh/common/dist/sandesh-common-0.1.dev0.tar.gz
+pushd sandesh-common-0.1.dev0
 %{__python} setup.py install --root=%{buildroot}  %{?_venvtr}
 popd
-tar zxf %{_build_dist}/tools/sandesh/library/python/dist/sandesh-0.1dev.tar.gz
-pushd sandesh-0.1dev
+tar zxf %{_build_dist}/tools/sandesh/library/python/dist/sandesh-0.1.dev0.tar.gz
+pushd sandesh-0.1.dev0
 %{__python} setup.py install --root=%{buildroot}  %{?_venvtr}
 popd
-tar zxf %{_build_dist}/discovery/dist/discovery-0.1dev.tar.gz
-pushd discovery-0.1dev
+tar zxf %{_build_dist}/discovery/dist/discovery-0.1.dev0.tar.gz
+pushd discovery-0.1.dev0
 %{__python} setup.py install --root=%{buildroot}  %{?_venvtr}
 popd
 
-tar zxf %{_build_dist}/discovery/client/dist/discoveryclient-0.1dev.tar.gz
-pushd discoveryclient-0.1dev
+tar zxf %{_build_dist}/discovery/client/dist/discoveryclient-0.1.dev0.tar.gz
+pushd discoveryclient-0.1.dev0
 %{__python}  setup.py install --root=%{buildroot} %{?_venvtr}
 popd
 
diff --git a/common/rpm/contrail-dependent-packages.spec b/common/rpm/contrail-dependent-packages.spec
index 34524f0a..74d5b84d 100644
--- a/common/rpm/contrail-dependent-packages.spec
+++ b/common/rpm/contrail-dependent-packages.spec
@@ -27,7 +27,7 @@
 
 Name:		    contrail-dependent-packages
 Version:	    %{_verstr}
-Release:	    %{_relstr}~%{_sku}%{?dist}
+Release:	    %{_relstr}%{?dist}
 Summary:	    Contrail Installer %{?_gitVer}
 BuildArch:          noarch
 
diff --git a/common/rpm/contrail-nodemgr.spec b/common/rpm/contrail-nodemgr.spec
index ab8b59c9..8a356e54 100644
--- a/common/rpm/contrail-nodemgr.spec
+++ b/common/rpm/contrail-nodemgr.spec
@@ -36,7 +36,7 @@ Requires:	  python-bottle >= 0.11.6
 Requires:	  python-contrail >= %{_verstr}-%{_relstr}
 Requires:	  ntp
 Requires:         python-psutil
-Requires:         PyYAML
+Requires:         python-PyYAML
 %if 0%{?rhel} && 0%{?rhel} <= 6
 Requires:         python-importlib
 %endif
@@ -94,8 +94,8 @@ popd
 mkdir -p build/python_dist
 pushd build/python_dist
 
-tar zxf %{_build_dist}/control-node/dist/Control-Node-0.1dev.tar.gz
-pushd Control-Node-0.1dev
+tar zxf %{_build_dist}/control-node/dist/Control-Node-0.1.dev0.tar.gz
+pushd Control-Node-0.1.dev0
 %{__python} setup.py install --root=%{buildroot}
 install -d -m 755 %{buildroot}/usr/share/doc/python-Control-Node
 if [ -d doc ]; then
@@ -103,8 +103,8 @@ if [ -d doc ]; then
 fi
 popd
 
-tar zxf %{_build_dist}/vnsw/agent/uve/dist/vrouter-0.1dev.tar.gz
-pushd vrouter-0.1dev
+tar zxf %{_build_dist}/vnsw/agent/uve/dist/vrouter-0.1.dev0.tar.gz
+pushd vrouter-0.1.dev0
 %{__python} setup.py install --root=%{buildroot}
 install -d -m 755 %{buildroot}/usr/share/doc/python-vrouter
 if [ -d doc ]; then
@@ -112,18 +112,18 @@ if [ -d doc ]; then
 fi
 popd
 
-tar zxf %{_build_dist}/opserver/node_mgr/dist/node_mgr-0.1dev.tar.gz
-pushd node_mgr-0.1dev
+tar zxf %{_build_dist}/opserver/node_mgr/dist/node_mgr-0.1.dev0.tar.gz
+pushd node_mgr-0.1.dev0
 %{__python} setup.py install --root=%{buildroot}
 popd
 
-tar zxf %{_build_dist}/nodemgr/dist/nodemgr-0.1dev.tar.gz
-pushd nodemgr-0.1dev
+tar zxf %{_build_dist}/nodemgr/dist/nodemgr-0.1.dev0.tar.gz
+pushd nodemgr-0.1.dev0
 %{__python} setup.py install --root=%{buildroot}
 popd
 
-tar zxvf %{_build_dist}/analytics/database/dist/database-0.1dev.tar.gz
-pushd database-0.1dev
+tar zxvf %{_build_dist}/analytics/database/dist/database-0.1.dev0.tar.gz
+pushd database-0.1.dev0
 %{__python} setup.py install --root=%{buildroot}
 popd
 
diff --git a/common/rpm/contrail-nova-vif.spec b/common/rpm/contrail-nova-vif.spec
index d6136e5a..619a447c 100644
--- a/common/rpm/contrail-nova-vif.spec
+++ b/common/rpm/contrail-nova-vif.spec
@@ -9,9 +9,11 @@
 %define         _verstr      1
 %endif
 
-%if 0%(grep -c Xen /etc/redhat-release)
-%define		dist	.xen
-%endif
+#%if 0%(grep -c Xen /etc/redhat-release)
+#%define		dist	.xen
+#%endif
+
+%define                dist    SUSE
 
 %{echo: "Building release %{_relstr}\n"}
 
@@ -41,5 +43,6 @@ popd
 %files
 %defattr(-,root,root,-)
 %{python_sitelib}/nova_contrail_vif*
+%{python_sitelib}/vif_plug_vrouter*
 
 
diff --git a/common/rpm/contrail-openstack-packages.spec b/common/rpm/contrail-openstack-packages.spec
index c6a7c213..c1329c29 100644
--- a/common/rpm/contrail-openstack-packages.spec
+++ b/common/rpm/contrail-openstack-packages.spec
@@ -27,7 +27,7 @@
 
 Name:		    contrail-openstack-packages
 Version:	    %{_verstr}
-Release:	    %{_relstr}~%{_sku}%{?dist}
+Release:	    %{_relstr}%{?dist}
 Summary:	    Contrail Installer %{?_gitVer}
 BuildArch:          noarch
 
diff --git a/common/rpm/contrail-openstack-vrouter.spec b/common/rpm/contrail-openstack-vrouter.spec
index f119e279..19052d32 100644
--- a/common/rpm/contrail-openstack-vrouter.spec
+++ b/common/rpm/contrail-openstack-vrouter.spec
@@ -27,9 +27,9 @@ BuildArch: noarch
 Requires: contrail-vrouter-common >= %{_verstr}-%{_relstr}
 
 # contrail-nova-vif not required from Kilo onwards
-%if %{_skuTag} == "havana" || %{_skuTag} == "icehouse" || %{_skuTag} == "juno"
-Requires: contrail-nova-vif >= %{_verstr}-%{_relstr}
-%endif
+#%if %{_skuTag} == "havana" || %{_skuTag} == "icehouse" || %{_skuTag} == "juno"
+#Requires: contrail-nova-vif >= %{_verstr}-%{_relstr}
+#%endif
 
 Requires: openstack-nova-compute
 Requires: openstack-utils
diff --git a/common/rpm/contrail-openstack-webui.spec b/common/rpm/contrail-openstack-webui.spec
index 53f508d7..43a620d9 100644
--- a/common/rpm/contrail-openstack-webui.spec
+++ b/common/rpm/contrail-openstack-webui.spec
@@ -37,7 +37,7 @@ Requires: contrail-web-core >= %{_verstr}-%{_relstr}
 Requires: contrail-web-controller >= %{_verstr}-%{_relstr}
 Requires: contrail-setup >= %{_verstr}-%{_relstr}
 Requires: contrail-utils >= %{_verstr}-%{_relstr}
-Requires: opscenter
+#Requires: opscenter
 
 %description
 Contrail Package Requirements for WebUI
diff --git a/common/rpm/contrail-packages.spec b/common/rpm/contrail-packages.spec
index 614c64b0..ee1e8f8c 100644
--- a/common/rpm/contrail-packages.spec
+++ b/common/rpm/contrail-packages.spec
@@ -28,7 +28,8 @@
 
 Name:		    contrail-packages
 Version:	    %{_verstr}
-Release:	    %{_relstr}~%{_sku}%{?dist}
+#Release:	    %{_relstr}~%{_sku}%{?dist}
+Release:	    %{_relstr}%{?dist}
 Summary:	    Contrail Installer %{?_gitVer}
 BuildArch:          noarch
 
diff --git a/common/rpm/contrail-setup.spec b/common/rpm/contrail-setup.spec
index f8a81dac..cf0e43d4 100644
--- a/common/rpm/contrail-setup.spec
+++ b/common/rpm/contrail-setup.spec
@@ -120,8 +120,8 @@ popd
 install -p -m 644 cfgm_utils.tgz  %{buildroot}%{_contrailopt}/cfgm_utils.tgz
 install -p -m 644 dns_scripts.tgz  %{buildroot}%{_contrailopt}/dns_scripts.tgz
 pushd %{_provdir}
-tar zxf dist/ContrailProvisioning-0.1dev.tar.gz
-cd ContrailProvisioning-0.1dev
+tar zxf dist/ContrailProvisioning-0.1.dev0.tar.gz
+cd ContrailProvisioning-0.1.dev0
 %{__python} setup.py install --root=%{buildroot} --install-scripts %{_contrailopt}/bin/
 popd
 
diff --git a/common/rpm/contrail-thirdparty-packages.spec b/common/rpm/contrail-thirdparty-packages.spec
index 105d1718..f59321fa 100644
--- a/common/rpm/contrail-thirdparty-packages.spec
+++ b/common/rpm/contrail-thirdparty-packages.spec
@@ -27,7 +27,7 @@
 
 Name:		    contrail-thirdparty-packages
 Version:	    %{_verstr}
-Release:	    %{_relstr}~%{_sku}%{?dist}
+Release:	    %{_relstr}%{?dist}
 Summary:	    Contrail Installer %{?_gitVer}
 BuildArch:          noarch
 
diff --git a/common/rpm/contrail-vrouter-common.spec b/common/rpm/contrail-vrouter-common.spec
index a76ee61d..1ed0305a 100644
--- a/common/rpm/contrail-vrouter-common.spec
+++ b/common/rpm/contrail-vrouter-common.spec
@@ -28,7 +28,7 @@ Vendor:             Juniper Networks Inc
 BuildArch: noarch
 
 Requires: contrail-vrouter >= %{_verstr}-%{_relstr}
-Requires: abrt
+#Requires: abrt
 # abrt-addon-vmcore might be needed for centos. Add when package is available
 %if 0%{?fedora} >= 17
 Requires: abrt-addon-vmcore
